javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -662,23 +662,40 @@ public T get(int index) {
     }
 
     @Override
-    public Array<T> dropUntil(Predicate<? super T> predicate) {
+    public Array<T> dropWhile(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
-        return dropWhile(predicate.negate());
+        return dropUntil(predicate.negate());
     }
 
     @Override
-    public Array<T> dropWhile(Predicate<? super T> predicate) {
+    public Array<T> dropUntil(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
         for (int i = 0; i < length(); i++) {
-            if (!predicate.test(get(i))) {
+            if (predicate.test(get(i))) {
                 return drop(i);
             }
         }
         return empty();
     }
 
     @Override
+    public Array<T> dropRightWhile(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return dropRightUntil(predicate.negate());
+    }
+
+    @Override
+    public Array<T> dropRightUntil(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        for (int i = length() - 1; i >= 0; i--) {
```

Comment:
`Array`, `Vector` and `CharSeq` are basically the same, should I extract it to `Collections`?